client/convert.go

```diff
@@ -0,0 +1,248 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/kubernetes/kompose/pkg/app"
+	"github.com/kubernetes/kompose/pkg/kobject"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func (k *Kompose) Convert(options ConvertOptions) ([]runtime.Object, error) {
+	options = k.setDefaultValues(options)
+	err := k.validateOptions(options)
+	if err != nil {
+		return nil, err
+	}
+	kobjectConvertOptions := kobject.ConvertOptions{
+		ToStdout:                    options.ToStdout,
+		CreateChart:                 k.createChart(options),
+		GenerateYaml:                true,
+		GenerateJSON:                options.GenerateJson,
+		Replicas:                    *options.Replicas,
+		InputFiles:                  options.InputFiles,
+		OutFile:                     options.OutFile,
+		Provider:                    k.getProvider(options),
+		CreateD:                     k.createDeployment(options),
+		CreateDS:                    k.createDaemonSet(options),
+		CreateRC:                    k.createReplicationController(options),
+		Build:                       *options.Build,
+		BuildRepo:                   k.buildRepo(options),
+		BuildBranch:                 k.buildBranch(options),
+		PushImage:                   options.PushImage,
+		PushImageRegistry:           options.PushImageRegistry,
+		CreateDeploymentConfig:      k.createDeploymentConfig(options),
+		EmptyVols:                   false,
+		Volumes:                     *options.VolumeType,
+		PVCRequestSize:              options.PvcRequestSize,
+		InsecureRepository:          k.insecureRepository(options),
+		IsDeploymentFlag:            k.createDeployment(options),
+		IsDaemonSetFlag:             k.createDaemonSet(options),
+		IsReplicationControllerFlag: k.createReplicationController(options),
+		Controller:                  k.getController(options),
+		IsReplicaSetFlag:            *options.Replicas != 0,
+		IsDeploymentConfigFlag:      k.createDeploymentConfig(options),
+		YAMLIndent:                  2,
+		WithKomposeAnnotation:       *options.WithKomposeAnnotations,
+		MultipleContainerMode:       k.multiContainerMode(options),
+		ServiceGroupMode:            k.serviceGroupMode(options),
+		ServiceGroupName:            k.serviceGroupName(options),
+		SecretsAsFiles:              k.secretsAsFiles(options),
+		GenerateNetworkPolicies:     options.GenerateNetworkPolicies,
+	}
+	err = app.ValidateComposeFile(&kobjectConvertOptions)
+	if err != nil {
+		return nil, err
+	}
+	objects, err := app.Convert(kobjectConvertOptions)
+	return objects, err
```

Comment:
I like the fact you return the object and then the user can convert that to yaml / json with their own library, as well as the error output. Good job!